qiskit_ibm_runtime/debug_tools/neat.py

```diff
@@ -0,0 +1,228 @@
+
+        # Run the Clifford PUBs on a QPU
+        r_qpu = estimator.run(cliff_pubs)
+
+        # Calculate useful figures of merit using mathematical operators, for example the relative
+        # difference between experimental and noisy results, ...
+        rel_diff = abs(r_noisy[0] - r_qpu[0]) / r_noisy[0]
+
+        # ... the signal-to-noise ratio between experimental and ideal results, ...
+        ratio = r_qpu[0] / r_ideal[0]
+
+        # ... or the absolute difference between results obtained with different noise models
+        abs_diff = abs(r_noisy[0] - another_r_noisy[0])
+
+    Args:
+        backend: A backend.
+        noise_model: A noise model for the operations of the given backend. If ``None``, it
+            defaults to the noise model generated by :meth:`NoiseModel.from_backend`.
+    """
+
+    def __init__(self, backend: Backend, noise_model: Optional[NoiseModel] = None) -> None:
+        if not HAS_QISKIT_AER:
+            raise ValueError(
+                "Cannot initialize object of type 'Neat' since 'qiskit-aer' is not installed. "
+                "Install 'qiskit-aer' and try again."
+            )
+
+        self._backend = backend
+        self.noise_model = (
+            noise_model
+            if noise_model is not None
+            else NoiseModel.from_backend(backend, thermal_relaxation=False)
+        )
+
+    @property
+    def noise_model(self) -> NoiseModel:
+        r"""
+        The noise model used by this analyzer tool for the noisy simulations.
+        """
+        return self._noise_model
+
+    @noise_model.setter
+    def noise_model(self, value: NoiseModel) -> NoiseModel:
+        """Sets a new noise model.
+
+        Args:
+            value: A new noise model.
+        """
+        self._noise_model = value
+
+    def backend(self) -> Backend:
+        r"""
+        The backend used by this analyzer tool.
+        """
+        return self._backend
+
+    def simulate(
+        self,
+        pubs: Sequence[EstimatorPubLike],
+        with_noise: bool = True,
+        seed_simulator: Optional[int] = None,
+        default_precision: float = 0,
+    ) -> NeatResult:
+        r"""
+        Calculates the expectation values for the estimator task specified by the given ``pubs``.
+
+        This function uses ``qiskit-aer``'s ``Estimator`` class to simulate the estimation task
+        classically.
+
+        .. note::
+            To ensure scalability, every circuit in ``pubs`` is required to be a Clifford circuit,
+            so that it can be simulated efficiently regardless of its size. For estimation tasks
+            that involve non-Clifford circuits, the recommended workflow consists of mapping
+            the non-Clifford circuits to the nearest Clifford circuits using the
+            :class:`.~ConvertISAToClifford` transpiler pass, or equivalently, to use the Neat's
+            :meth:`to_clifford` convenience method.
```

Comment:
Could we simplify the workflow by adding a kwarg for automatic conversion for the lazy folk? I haven't thought too hard about the consequences.

```python
    def simulate(
        self,
        pubs: Sequence[EstimatorPubLike],
        with_noise: bool = True,
        cliffordize: bool = True,
        seed_simulator: Optional[int] = None,
        default_precision: float = 0,
    ) -> NeatResult:
```

Reply:
I have thought about this. The consequence I see is that people may forget about it, end end up doing the following thing:
```
# run non-Cliff pubs on CPU
results = estimator.run(pubs).results()

# debugger (which automatically turns PUBs into Clifford PUBs)
results_deb = Neat(backend).simulate(pubs)

# compare
ratio = results / results_deb
```
So at the very least, if we want to do this, we should set `cliffordize` to `False` by default